src/lib/components/commons/FilterPanel/FilterPanel.tsx

```diff
@@ -0,0 +1,132 @@
+
+  const form = useForm<ProjectFormValues>({
+    resolver: zodResolver(projectFormSchema),
+    defaultValues: PROJECT_FORM_DEFAULT_VALUES,
+    mode: 'onSubmit',
+  });
+
+  return (
+    <>
+      <SvgIcon
+        size='S'
+        onClick={() => {
+          setShowFilters(!showFilters);
+        }}
+      >
+        <use
+          href={`/img/icons.svg#icon-filter-${showFilters ? 'open' : 'close'}`}
+        ></use>
+      </SvgIcon>
+
+      <FormProvider {...form}>
+        {showFilters && (
+          <StyledFilterPanel>
+            <StyledFilterControlDiv>
+              <p style={{ flexBasis: '100%' }}>
+                {PROJECT_DISPLAY_TEXTS.he.fields.status}
+              </p>
+              <StyledFilterButton
+                isButtonGroup={true}
+                variant={
+                  status === 'cancel'
+                    ? 'secondary'
+                    : status
+                    ? 'primary'
+                    : 'secondary'
+                }
+                onClick={() =>
+                  setStatus(!status ? true : status === true ? 'cancel' : true)
+                }
+              >
+                פעיל
+              </StyledFilterButton>
+              <StyledFilterButton
+                isButtonGroup={true}
+                variant={
+                  status === 'cancel'
+                    ? 'secondary'
+                    : !status
+                    ? 'primary'
+                    : 'secondary'
+                }
+                onClick={() =>
+                  setStatus(
+                    status ? false : status === false ? 'cancel' : false,
+                  )
+                }
+              >
+                לא פעיל
+              </StyledFilterButton>
+            </StyledFilterControlDiv>
+
+            <StyledFilterControlDiv>
+              <p style={{ flexBasis: '100%' }}>
+                {PROJECT_DISPLAY_TEXTS.he.fields.sDate}
+              </p>
+              <DateInput
+                label={'מ'}
+                name={`${ProjectFields.SDate} sDateFilter`}
+              ></DateInput>
+              <DateInput
+                label={'עד'}
+                name={`${ProjectFields.EDate} sDateFilter`}
+              ></DateInput>
+            </StyledFilterControlDiv>
+
+            <StyledFilterControlDiv>
+              <p style={{ flexBasis: '100%' }}>
+                {PROJECT_DISPLAY_TEXTS.he.fields.eDate}
+              </p>
+              <DateInput
+                label={'מ'}
+                name={`${ProjectFields.SDate} eDate`}
+              ></DateInput>
+              <DateInput
+                label={'עד'}
+                name={`${ProjectFields.EDate} eDate`}
+              ></DateInput>
+            </StyledFilterControlDiv>
+            <StyledFilterControlDiv justify='flex-end'>
+              <StyledFilterButton
+                width='25%'
+                variant='primary'
+                onClick={() => setShowFilters(!showFilters)}
```

Comment:
use callback version as mentioned above. and for sure you can make a toggle function as well..
BTW , you can create custom hook for this (e.g.):
```
export const useToggle(initState:boolean)=>{
  const [state, setState] = useState(initState)
  const toggle = ()=>setState(prev=>!prev)
  return [state , toggle]
}
```